preprocessing/meshing/cube_c/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+cmake_minimum_required(VERSION 3.10)
+
+# set the project name
+project(cube_c)
+
+# specify the C++ standard
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+cmake_policy(SET CMP0074 NEW)
+
+
+add_executable(cubeGenerator src/main.cpp)
+
+#logging
+set(LOG_LEVEL "warning" CACHE STRING "Log level for the code")
+set(LOG_LEVEL_OPTIONS "debug" "info" "warning" "error")
+set_property(CACHE LOG_LEVEL PROPERTY STRINGS ${LOG_LEVEL_OPTIONS})
+if("${LOG_LEVEL}" STREQUAL "debug")
+  target_compile_definitions(cubeGenerator PUBLIC LOG_LEVEL=3)
+elseif("${LOG_LEVEL}" STREQUAL "info")
+  target_compile_definitions(cubeGenerator PUBLIC LOG_LEVEL=2)
+elseif("${LOG_LEVEL}" STREQUAL "warning")
+  target_compile_definitions(cubeGenerator PUBLIC LOG_LEVEL=1)
+elseif("${LOG_LEVEL}" STREQUAL "error")
+  target_compile_definitions(cubeGenerator PUBLIC LOG_LEVEL=0)
+endif()
+
+#build and link libraries and executable
+set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
```

Comment:
Imo it's better to set this to `set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/../../../cmake" ${CMAKE_MODULE_PATH})` and delete the symlink.